tests/model_registry/python_client/test_model_registry_creation.py

```diff
@@ -86,6 +70,49 @@ def test_model_registry_operator_env(
         if not namespace_env:
             pytest.fail("Missing environment variable REGISTRIES_NAMESPACE")
 
+    def test_multiple_registry_clients_with_factory(
+        self,
+        default_model_registry_factory,
+        model_registry_client_factory,
+    ):
+        """Test using factory fixtures to create multiple registry instances and clients."""
+        LOGGER.info("Testing multiple Model Registry instances with client factory")
+
+        # Create two different registry instances
```

Comment:
I don't like this. This is changing a typical expectation of how pytest works. Typically setups/teardown's are part of fixtures and tests does the validations. Here we are blurring the lines inorder to reduce code duplications, but introducing complexity, that can make triage problematic. I would prefer a simple solution keeping in mind zen of python. If no simplifications are possible, then let's log enhancements for pytest.